lib/util/exec/docker/index.ts

```diff
@@ -198,13 +198,17 @@ export async function generateDockerCommand(
 
   result.push(...prepareVolumes([localDir, cacheDir, ...volumes]));
 
+  const customEnvVariables = config.customEnvVariables || {};
   if (envVars) {
     result.push(
       ...uniq(envVars)
-        .filter((x) => typeof x === 'string')
+        .filter((x) => typeof x === 'string' && !customEnvVariables[x])
```

Comment:
```suggestion
        .filter((x) => typeof x === 'string' && customEnvVariables[x] === undefined)
```

Is there a possibility that a custom env variable could be "falsy"? e.g. `false`, `0`

Reply:
Indeed. Good one!